server/src/test/java/org/opensearch/index/mapper/KeywordFieldTypeTests.java

```diff
@@ -197,6 +205,57 @@ public void testTermsQuery() {
         );
     }
 
+    public void testTermsSortedQuery() {
+        String[] seedStrings = generateRandomStringArray(10, 10, false, true);
+        List<BytesRef> bytesRefList = Arrays.stream(seedStrings).map(BytesRef::new).toList();
+        List<String> sortedStrings = bytesRefList.stream().sorted().map(BytesRef::utf8ToString).toList();
+
+        MappedFieldType ft = new KeywordFieldType("field");
+        Query expected = new IndexOrDocValuesQuery(
+            new TermInSetQuery("field", bytesRefList),
+            new TermInSetQuery(MultiTermQuery.DOC_VALUES_REWRITE, "field", bytesRefList)
+        );
+        assertEquals(expected, ft.termsQuery(sortedStrings, MOCK_QSC_ENABLE_INDEX_DOC_VALUES));
+
+        MappedFieldType onlyIndexed = new KeywordFieldType("field", true, false, Collections.emptyMap());
+        Query expectedIndex = new TermInSetQuery("field", bytesRefList);
+        assertEquals(expectedIndex, onlyIndexed.termsQuery(sortedStrings, null));
+
+        MappedFieldType onlyDocValues = new KeywordFieldType("field", false, true, Collections.emptyMap());
+        Query expectedDocValues = new TermInSetQuery(MultiTermQuery.DOC_VALUES_REWRITE, "field", bytesRefList);
+        assertEquals(expectedDocValues, onlyDocValues.termsQuery(sortedStrings, null));
+    }
+
+    @AwaitsFix(bugUrl = "never.ever")
+    public void testMockTermsSortedQuery() {
```

Comment:
how to test constructor argument type? 
sadly, I need to remove this test. 
let me know if you have an idea.

Reply:
sorry, can you provide some more details on what exactly you're trying to test here?

Reply:
the key thing here is `assertTrue(args[i] instanceof SortedSet<?>);` but `mockConstructionWithAnswer` requires `mockito-inline` dependency which I'm not able to configure. I suppose I'll just remove this test. Before commit.